src/coreclr/jit/morph.cpp

```diff
@@ -13862,53 +13890,16 @@ void Compiler::fgMorphBlocks()
         fgEnsureFirstBBisScratch();
     }
 
-    /*-------------------------------------------------------------------------
-     * Process all basic blocks in the function
-     */
-
-    BasicBlock* block = fgFirstBB;
-    noway_assert(block);
-
-    do
+    // Morph all blocks.
+    //
+    // Note morph can add blocks downstream from the current block,
+    // and alter (but not null out) the current block's bbNext;
+    // this iterator ensures they all get visited.
```

Comment:
Doesn't the iterator load the bbNext before the iteration of the loop? If the callee can adjust the blocks, maybe we shouldn't use the for-based iterator here?

Reply:
It doesn't seem to cache anything...? Also that comment is a bit stale; with other changes I've made recently morph won't add blocks or (I suspect) alter `bbNext` anymore.

Reply:
Maybe it only caches the end-point of the iteration. But still, any change to the block list seems risky to depend on what the C++ compiler is going to do.

Reply:
For the `BasicBlockRangeList` form sure... but this is going via `BasicBlockSimpleList` which does't cache an endpoint.

Reply:
@BruceForstall still ok with merging? Or would you rather I change the iteration back?

Reply:
Yes, it's fine.